operator/eni.go

```diff
@@ -153,9 +153,9 @@ func startENIAllocator(awsClientQPSLimit float64, awsClientBurst int, eniTags ma
 		mngr.UpdateController("eni-refresh",
 			controller.ControllerParams{
 				RunInterval: time.Minute,
-				DoFunc: func(_ context.Context) error {
-					syncTime := instances.Resync()
-					nodeManager.Resync(syncTime)
+				DoFunc: func(ctx context.Context) error {
```

Comment:
Whose responsibility is it to check whether the controller deadline was exceeded? Will the `ControllerManager` do this or is this function supposed to return `ctx.Err()`?

Reply:
The context that is passed into the `DoFunc` will be one of the following:

* the context passed into the `ControllerParams` (not passed here).
* the context created for this controller (created from `context.Background()`)

The controllers do not use deadlines, rather, they are created using `WithCancel`; the controller is canceled when it is stopped. I was thinking in the future, that when the operator is stopped / terminated, we would cancel all ongoing operations to tear down gracefully. This is not done here yet. I could use `context.TODO()` here instead for now and file a follow-up issue?

Reply:
I think that this code is fine as-is. The follow-up issue for cancelling ongoing operations to tear down gracefully would be nice though, yeah. Thanks.

Reply:
Created https://github.com/cilium/cilium/issues/9529